server/etcdserver/bootstrap.go

```diff
@@ -240,42 +331,43 @@ func bootstrapNewClusterNoWAL(cfg config.ServerConfig, prt http.RoundTripper, st
 			return nil, err
 		}
 	}
-	cl.SetStore(st)
-	cl.SetBackend(schema.NewMembershipBackend(cfg.Logger, be))
-	br := bootstrapRaftFromCluster(cfg, cl, cl.MemberIDs())
-	cl.SetID(br.wal.id, cl.ID())
-	return &bootstrappedServer{
+	return &bootstrapedCluster{
 		remotes: nil,
-		raft:    br,
+		cl:      cl,
+		nodeID:  m.ID,
 	}, nil
 }
 
-func bootstrapWithWAL(cfg config.ServerConfig, st v2store.Store, be backend.Backend, ss *snap.Snapshotter, beExist bool, beHooks *serverstorage.BackendHooks, ci cindex.ConsistentIndexer) (*bootstrappedServer, error) {
+func bootstrapClusterWithWAL(cfg config.ServerConfig, meta *snapshotMetadata) (*bootstrapedCluster, error) {
 	if err := fileutil.IsDirWriteable(cfg.MemberDir()); err != nil {
 		return nil, fmt.Errorf("cannot write to member directory: %v", err)
 	}
 
-	if err := fileutil.IsDirWriteable(cfg.WALDir()); err != nil {
-		return nil, fmt.Errorf("cannot write to WAL directory: %v", err)
-	}
-
 	if cfg.ShouldDiscover() {
 		cfg.Logger.Warn(
 			"discovery token is ignored since cluster already initialized; valid logs are found",
-			zap.String("wal-dir", cfg.WALDir()),
+			zap.String("bwal-dir", cfg.WALDir()),
```

Comment:
This refers to bootstraped with WAL dir right? Would it be okay if we said "bootstrap-wal-dir" as it might not be obvious to everyone, what do you think?

Reply:
I think that was also result of intelij rename "wal" -> "bwal". Sorry for missing it.